src/output-model.js

```diff
@@ -0,0 +1,93 @@
+
+import {EventEmitter} from 'events';
+
+/**
+ * Provides Jupyter outputs to an output area.
+ *
+ * This class manages the conversion of Jupyter messages to output state, and
+ * handles adding the output state to the output area.
+ */
+export class OutputModel extends EventEmitter {
+    
+    
+    /**
+     * Public constructor
+     */
+    constructor() {
+        super();
+        this.clearQueued = false;
+        this._state = [];        
```

Comment:
I may make a PR later to hide `_state` with a closure since we always use the getter and setter. I should probably wait for tests before I do that.